aws/rust-runtime/aws-runtime/src/auth.rs

```diff
@@ -199,7 +209,8 @@ fn extract_field_from_endpoint_config<'a>(
         .and_then(|config| config.get(field_name))
 }
 
-fn apply_signing_instructions(
+/// Applies the instructions to the given `request`.
+pub fn apply_signing_instructions(
```

Comment:
I attempted to use `signing_instructions.apply_to_request_http0x(request)` in `S3ExpressSigner::sign_http_request`, but the variable `request` was passed by a caller whose type is `&mut HttpRequest`.
Calling  `.try_into_http02x()` on `request` within `S3ExpressSigner::sign_http_request` would convert that to the underlying type, but `request` needed to continue being used, expecting it to be orchestrator's `HttpRequest`.

That's the rationale for making this function `pub`.